src/main/java/duke/DateTimeParser.java

```diff
@@ -0,0 +1,127 @@
+     * @throws DateTimeParseException if dateTimeString does not have an accepted format
+     */
+    public static LocalDateTime parse(String dateTimeString) throws DateTimeParseException {
+        return LocalDateTime.from(parser.parse(dateTimeString));
+    }
+
+    /**
+     * Returns a String version of the specified date in a format suitable for storage.
+     *
+     * @param date a LocalDate object containing the date to be converted into a string.
+     * @return String version of the specified date
+     * @throws DateTimeException
+     */
+    public static String dateToStorageString(LocalDate date) throws DateTimeException {
+        return dateStorageFormatter.format(date);
+    }
+
+    /**
+     * Returns a String version of the specified date and time in a format suitable for storage.
+     *
+     * @param dateTime a LocalDateTime object containing the date and time to be converted into a string.
+     * @return String version of the specified date and time
+     * @throws DateTimeException
+     */
+    public static String dateTimeToStorageString(LocalDateTime dateTime) throws DateTimeException {
+        return dateTimeStorageFormatter.format(dateTime);
+    }
+
+    /**
+     * Returns a String version of the specified date in a format suitable for display.
+     *
+     * @param date a LocalDate object containing the date to be converted into a string.
+     * @return String version of the specified date
+     * @throws DateTimeException
+     */
+    public static String dateToDisplayString(LocalDate date) throws DateTimeException {
+        String formattedDate = dateDisplayFormatter.format(date);
+        int dayOfTheMonth = date.getDayOfMonth();
+        String ordinal = getOrdinal(dayOfTheMonth);
+        return dayOfTheMonth + ordinal + " " + formattedDate;
+    }
+
+    /**
+     * Returns a String version of the specified date and time in a format suitable for display.
+     *
+     * @param dateTime a LocalDateTime object containing the date to be converted into a string.
+     * @return String version of the specified date and time
+     * @throws DateTimeException
+     */
+    public static String dateTimeToDisplayString(LocalDateTime dateTime) throws DateTimeException {
+        String formattedDate = dateDisplayFormatter.format(dateTime);
+
+        int dayOfTheMonth = dateTime.getDayOfMonth();
+        String ordinal = getOrdinal(dayOfTheMonth);
+
+        int minuteOfHour = dateTime.getMinute();
+        String minutes = (minuteOfHour == 0) ? "" : (":" + minuteOfHour);
+
+        int hourWithAmPm = dateTime.get(ChronoField.CLOCK_HOUR_OF_AMPM);
+        String amPm = getAmPm(dateTime.getHour());
+
+        return dayOfTheMonth + ordinal + " " + formattedDate + ", " + hourWithAmPm + minutes + amPm;
+    }
+
+    private static String getOrdinal(final int day) {
+        if (day >= 11 && day <= 13) {
+            return "th";
+        }
+        switch (day % 10) {
+        case 1: return "st";
```

Comment:
Minor nit here but the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#layout) does seem to require an explicit `// Fallthrough` comment whenever there is a switch case without a break statement.